tools/circt-rtl-sim/circt-rtl-sim.py.in

```diff
@@ -79,6 +79,60 @@ class Questa:
         return subprocess.run(cmd + simargs.split())
 
 
+class Vivado:
+    """Run and compile funcs for Vivado."""
+
+    DefaultDriver = "driver.sv"
+
+    def __init__(self, path, args):
+        self.path = path
+        self.args = args
+
+    def compile(self, sources):
+        sources = filter(
+            lambda fn: not (fn.endswith(".so") or fn.endswith(".dll")),
+            sources)
+        vlog = os.path.join(self.path, "xvlog")
+        args = [vlog, "-sv"] + list(sources)
+        rc = subprocess.run(args)
+        if rc.returncode != 0:
+            return rc
+
+        elab = os.path.join(self.path, "xelab")
+        if self.args.no_default_driver:
+            top = self.args.top
+        else:
+            top = "driver"
+        args = [elab, top, "-debug", "typical"]
+        return subprocess.run(args)
+
+    def run(self, cycles, simargs):
+        xsim = os.path.join(self.path, "xsim")
+
+        if self.args.no_default_driver:
+            top = self.args.top
+        else:
+            top = "driver"
+
+        args = [xsim, top]
+
+        script = "script.tcl"
```

Comment:
Does 'xvsim' not support `-c 'run -all'` on the command line? That's what I do for Questa and it'd be nice to keep them consistent.

Reply:
Nevermind... Just saw your comment.

Reply:
Not that I saw. There is a `-R`, which is like `run -all; quit`. I didn't want to use that because I wanted this script to support interactive use through the `--gui` option, and `-R` would open the gui, "run all", but then close the gui.